### ISSUE TITLE
[BUG] ext4-fs error salvaged volume issue

### ISSUE THREAD

**Issue by albertkohl-monotek** [harvester/harvester#3238]
#**Describe the bug**
I'm not sure what happened, but serveral of my VMs had an IO error. one of them presented with an ext-fs error. (ext4-fs error device dm-0 ext4_lookup inoode spamming the console)
i restarted the vm, came up w/ the same error in the console.

i then shut down all vms, and put the node into Maintenance Mode, and rebooted the node.

now the vm wont start at all, and i see this in the events for the volume in longhorn.

<html>
<body>
<br class="Apple-interchange-newline">

Name | Kind | Source | Type | First Time | Last Time | Reason | Message
-- | -- | -- | -- | -- | -- | -- | --
pvc-a93f56c7-0b5c-4ff9-a59b-6a0cdce0ac17-r-a740e095 | Replica | longhorn-replica-controller | Normal | 41 minutes ago | a few seconds ago | Start | Starts pvc-a93f56c7-0b5c-4ff9-a59b-6a0cdce0ac17-r-a740e095
pvc-a93f56c7-0b5c-4ff9-a59b-6a0cdce0ac17 | Volume | longhorn-volume-controller | Normal | 31 minutes ago | 31 minutes ago | Detached | volume pvc-a93f56c7-0b5c-4ff9-a59b-6a0cdce0ac17 has been detached
pvc-a93f56c7-0b5c-4ff9-a59b-6a0cdce0ac17 | Volume | longhorn-volume-controller | Normal | 3 hours ago | 42 minutes ago | Detached | volume pvc-a93f56c7-0b5c-4ff9-a59b-6a0cdce0ac17 has been detached
pvc-a93f56c7-0b5c-4ff9-a59b-6a0cdce0ac17 | Volume | longhorn-volume-controller | Warning | 3 hours ago | an hour ago | AutoSalvaged | Replica pvc-a93f56c7-0b5c-4ff9-a59b-6a0cdce0ac17-r-a740e095 of volume pvc-a93f56c7-0b5c-4ff9-a59b-6a0cdce0ac17 will be automatically salvaged
pvc-a93f56c7-0b5c-4ff9-a59b-6a0cdce0ac17 | Volume | longhorn-volume-controller | Normal | an hour ago | an hour ago | Remount | Volume pvc-a93f56c7-0b5c-4ff9-a59b-6a0cdce0ac17 requested remount at 2022-12-03T00:52:15Z after automatically salvaging replicas
pvc-a93f56c7-0b5c-4ff9-a59b-6a0cdce0ac17 | Volume | longhorn-volume-controller | Normal | an hour ago | an hour ago | Remount | Volume pvc-a93f56c7-0b5c-4ff9-a59b-6a0cdce0ac17 requested remount at 2022-12-03T00:52:14Z after automatically salvaging replicas
pvc-a93f56c7-0b5c-4ff9-a59b-6a0cdce0ac17 | Volume | longhorn-volume-controller | Normal | an hour ago | an hour ago | Remount | Volume pvc-a93f56c7-0b5c-4ff9-a59b-6a0cdce0ac17 requested remount at 2022-12-03T00:52:10Z after automatically salvaging replicas
pvc-a93f56c7-0b5c-4ff9-a59b-6a0cdce0ac17 | Volume | longhorn-volume-controller | Normal | an hour ago | an hour ago | Remount | Volume pvc-a93f56c7-0b5c-4ff9-a59b-6a0cdce0ac17 requested remount at 2022-12-03T00:52:08Z after automatically salvaging replicas
pvc-a93f56c7-0b5c-4ff9-a59b-6a0cdce0ac17 | Volume | longhorn-volume-controller | Normal | an hour ago | an hour ago | Remount | Volume pvc-a93f56c7-0b5c-4ff9-a59b-6a0cdce0ac17 requested remount at 2022-12-03T00:52:06Z after automatically salvaging replicas
pvc-a93f56c7-0b5c-4ff9-a59b-6a0cdce0ac17 | Volume | longhorn-volume-controller | Normal | an hour ago | an hour ago | Remount | Volume pvc-a93f56c7-0b5c-4ff9-a59b-6a0cdce0ac17 requested remount at 2022-12-03T00:52:04Z after automatically salvaging replicas

</body>
</html>

anthing i can do to bring the volume up and attempt to repair / salvage some volumes would be helpful. 

**To Reproduce**
not sure what caused the issue, some sort of i/o disruption

**Expected behavior**
node should fail over properly to one of the other replicas, but for some reason, i show only the single replica. (should have 3)

**Support bundle**
attached

**Environment**
v1.0.3. i'm getting ready to upgrade to v1.1.1 but want to ensure everything is ready before i do
bare metal Dell R620s

**Additional context**
had a similar issue a couple weeks ago with same node but different issue.
[[BUG] DAS - Partitions lost](https://github.com/harvester/harvester/issues/3128)
#3128

**Comment by Martin-Weiss** [comment]
There has been a potential risk of corruption in longhorn in harvester up to 1.0.3 which has been fixed in longhorn 1.3.2 which is part of harvester 1.1.0 and newer. Maybe this is what has been causing this problem, here..

**Comment by Vicente-Cheng** [comment]
Hi @albertkohl-monotek, 
Is any migration or node temporary down on your environment?

v1.0.3 has a potential risk for data corruption issues in longhorn, but it only occurs on volume replica rebuild.
So I want to confirm the above situation, which would cause a replica rebuild.
Thanks!

**Comment by albertkohl-monotek** [comment]
@Vicente-Cheng no, nothing went "down" i'm guessing there was some sort of "hiccup" with that node, it's the same one you helped me with the JBOD disconnecting a couple weeks ago. once it dug into it, longhorn has only the one replica. that seems to be the symtom of longhorn curruption. https://github.com/longhorn/longhorn/issues/4354

i just rebuild the vm from backup, and moved on. little data loss, but i'll live.

i do want to upgrade my cluster from v1.0.3 to v.1.1.1 to prevent this issue in the future (due to the longhorn upgrade), are their any pre-upgrade tasks i can run to ensure a smooth upgrade? this is a production cluster, so i dont wanna have issues with it hanging up.

**Comment by Vicente-Cheng** [comment]
Hi @albertkohl-monotek,
OK, thanks for your information.

For the JBOD disconnection issue, I open another issue to track (#3218).
There is something we could handle better on the harvester side.

For the pre-upgrade tasks.
You could ensure the VM status is OK and shut it down before the upgrade. That would reduce the rebuild operation on the `Longhorn` side and prevent live migration by `kubevirt`.
After that, we still need to confirm that all volume statuses are healthy. (You could check it on longhorn UI)
We introduce some checking mechanisms (to make your data safer). If you start to upgrade w/o a healthy volume, it may be stuck on some checks.

I thought the upgrade would be smoother if the above checking was ready.
There are still some known issues with the upgrade. Refer to this page https://docs.harvesterhci.io/v1.1/upgrade/v1-0-3-to-v1-1-1.

If you have any upgrade issues, feel free to open an issue here. We will help resolve it.
Thanks!

**Comment by albertkohl-monotek** [comment]
@Vicente-Cheng question, you mentioned: 

> After that, we still need to confirm that all volume statuses are healthy. (You could check it on longhorn UI)

if i have a two node cluster (staging cluster) and the volumes are degraded because i only have 2 of 3 replicas, will this block the upgrade? 

also, if it hangs, can i still take backups? (so i can still backup/restore to another cluster if i get stuck)

**Comment by Vicente-Cheng** [comment]
> @Vicente-Cheng question, you mentioned:
> 
> > After that, we still need to confirm that all volume statuses are healthy. (You could check it on longhorn UI)
> 
> if i have a two node cluster (staging cluster) and the volumes are degraded because i only have 2 of 3 replicas, will this block the upgrade?
> 
> also, if it hangs, can i still take backups? (so i can still backup/restore to another cluster if i get stuck)

hi @albertkohl-monotek, 
If you only have two nodes in the cluster, maybe adjust the replica number to 2. Or the upgrade would hang on waiting volume healthy.

I thought we could do the backup before the upgrade instead of hanging the upgrade.

**Comment by albertkohl-monotek** [comment]
@Vicente-Cheng my hope is to not have to "backup" every single vm, i have about 10 TB of data if i had to do it that way. i was thinking i can basically test the upgrade with some of the vms from the test cluster, configured similar to the prod one, if it upgrades with no issues, then i can upgrade the prod cluster. My question is more around, if the the test cluster works, but the prod one still fails, can i do a backup / restore from prod -> test. then my test can just replace the prod. does that make sense? 

or will my whole prod-cluster be high-and-dry if i start the upgrade, and it dies during the process?
